docs/guide/coverage.md

```diff
@@ -88,3 +88,24 @@ export default defineConfig({
 ```
 
 Please refer to the type definition for more details.
+
+## Ignoring code
+
+Both coverage providers have their own ways how to ignore code from coverage reports.
+
+- `c8`: https://github.com/bcoe/c8#ignoring-uncovered-lines-functions-and-blocks
+- `ìstanbul` https://github.com/istanbuljs/nyc#parsing-hints-ignoring-lines
+
+When using Typescript the source codes are transpiled using `esbuild`, which strips all comments from the source codes ([esbuild#516](https://github.com/evanw/esbuild/issues/516)).
+Comments which are considered as [legal comments](https://esbuild.github.io/api/#legal-comments) are preserved.
+
+For `istanbul` provider you can include a `@preserve` keyword in the ignore hint.
+Beware that these ignore hints may now be included in final production build as well.
+
+```diff
+-/* istanbul ignore if */
++/* istanbul ignore if -- @preserve */
+if (condition) {
+```
+
+Unfortunately this does not work for `c8` at the moment.
```

Comment:
The reason is that these regexps simply don't recognize comments if there is something between the keyword and closing tag.
https://github.com/istanbuljs/v8-to-istanbul/blob/b5ecd8242cf64389c9a656d78530bd0428edf195/lib/source.js#L54-L69

Reply:
Upstream PR open: https://github.com/istanbuljs/v8-to-istanbul/pull/203